app/views/markets/markets.html.erb

```diff
@@ -0,0 +1,48 @@
+<%= link_to "<< return to main page", "/" %>
+<h1>All Markets</h1>
+<%= form_tag({controller:'markets', action: 'search'}, method: "get") do %>
+<%= text_field_tag :search, params[:search], id: 'search_field', placeholder: "Search by Name" %>
+<%= submit_tag "Search" %>
+<% end %>
+<p>
+  <%= link_to "+ Add a Market", new_market_path %>
+</p>
+<table>
+  <thead>
+    <tr>
+      <th>
+        Name <%= link_to "&#8595".html_safe, "/markets_/name_sort" %>
```

Comment:
This should be using markets_name_sort_path rather than a direct link.  

Using `rake routes` can be really helpful to figure out what routes you have available: ```markets_name_sort   GET   /markets/name_sort(.:format)```

Refactor to the following to get the search and sort functionality working.
```Name <%= link_to "&#8595".html_safe, markets_sort_path(:search => params[:search], :sort => "name")%>```